hibernate-core/src/main/java/org/hibernate/sql/ast/tree/select/QuerySpec.java

```diff
@@ -112,6 +121,41 @@ public void accept(SqlAstWalker sqlTreeWalker) {
 
 	@Override
 	public MappingModelExpressable getExpressionType() {
-		return null;
+		SqlSelection first = selectClause.getSqlSelections().get(0);
+		return ( (SqlSelectionImpl) first ).getWrappedSqlExpression().getExpressionType();
```

Comment:
@sebersole I needed this typecast in three places. I guess either:

- `getWrappedSqlExpression()` should be pulled up to the interface, or
- `we should add a `getExpressionType()` method to the interface.

Or something else?

Reply:
Probably the second makes more sense

Reply:
OK, I will make that change.

Reply:
> OK, I will make that change.

Done.

Reply:
Nice!